tensorflow/contrib/solvers/python/ops/linear_equations.py

```diff
@@ -84,15 +85,15 @@ def conjugate_gradient(operator,
 
   def stopping_criterion(i, state):
     return math_ops.logical_and(i < max_iter,
-                                util.l2norm_squared(state.r) > tol)
+                                linalg_ops.norm(state.r) > tol)
 
   def cg_step(i, state):
     z = operator.apply(state.p)
     alpha = state.gamma / util.dot(state.p, z)
     x = state.x + alpha * state.p
     r = state.r - alpha * z
     if preconditioner is None:
-      gamma = util.l2norm_squared(r)
+      gamma = util.dot(r, r)
       beta = gamma / state.gamma
```

Comment:
using `util.dot()` here because `linalg_ops.norm()` apply `sqrt` , we have to call `square` to get squared_norm, this could cause precision slightly lower than `dot()` every step, we want to avoid it when iter_num is large. 
In fact using  `linalg_ops.norm(r) ** 2` here would make results "much" different when preconditioner=`None` and `identity_operator`(which supposed to be identical).

Reply:
OK, that's fine.